service/selectors/entities/files.js

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {createSelector} from 'reselect';
+
+function getAllFiles(state) {
+    return state.entities.files.files;
+}
+
+function getFilesIdsForPost(state, props) {
+    return state.entities.files.fileIdsByPostId[props.post.id] || [];
```

Comment:
See my last comment. This could instead be either `return props.post.file_ids || [];` or `return state.entities.posts.posts[props.post.id] || [];`